spec/features/consumer/shopping/cart_spec.rb

```diff
@@ -20,6 +20,17 @@
       set_order order
     end
 
+    it "shows the right registration link" do
+      # We have an issue with the registration link within Spree controllers.
+      # The `registration_path` helper resolves to `/signup` due to
+      # spree_auth_device > config > routes.rb. This spec verifies that we have
+      # a solution to that problem.
+      add_product_to_cart order, product_with_fee
+      visit spree.cart_path
+      register_page = window_opened_by { click_link "Register here" }
+      within_window(register_page) { expect(page).to have_content "Welcome" }
+    end
```

Comment:
It feels a bit too much to boot a browser to only check if that a link points to `/register`. Since in this case, we don't care about the user input or interaction involving different pages but the target of a link why don't we try using view specs? What are your thoughts?

Reply:
I'm not 100% sure but I don't think the problem is with the view. I think the problem is with the controller and which helpers it has loaded. A controller spec may work. I can give that a go and see if we can reproduce the bug that way. Is it worth the dev time?

Reply:
@sauloperez Please have a look at my new commit. I replaced the feature spec with two controller specs. It saves us three seconds and it's much more specific, closer to the problem.